unittest/gunit/keyring/keys_container-t.cc

```diff
@@ -273,7 +273,367 @@ namespace keyring__keys_container_unittest
+    ASSERT_TRUE(key_data_with_version.length()+1 == key_data_fetched_size);
+
+    my_free(fetched_key->release_key_data());
+    my_free(fetched_key_2->release_key_data());
+    my_free(fetched_key_3->release_key_data());
+    my_free(fetched_sk->release_key_data());
+
+    delete sample_key; // unused in this test
+  }
+
+  TEST_F(Keys_container_test, StoreStoreStoreSystemKeyAndTryRemovingSystemKey)
+  {
+    IKeyring_io *keyring_io= new Buffered_file_io(logger);
+    EXPECT_EQ(keys_container->init(keyring_io, file_name), 0);
+
+    std::string key_data1("system_key_data_1");
+    Key *key1= new Key("percona_binlog:0", "AES", NULL, key_data1.c_str(), key_data1.length()+1);
+
+    EXPECT_EQ(keys_container->store_key(key1), 0);
+    ASSERT_TRUE(keys_container->get_number_of_keys() == 1);
+
+    std::string key_data2("system_key_data_2");
+    Key *key2= new Key("percona_binlog:1", "AES", NULL, key_data2.c_str(), key_data2.length()+1);
+    EXPECT_EQ(keys_container->store_key(key2), 0);
+    ASSERT_TRUE(keys_container->get_number_of_keys() == 2);
+
+    std::string key_data3("system_key_data_3");
+    Key *key3= new Key("percona_binlog:2", "AES", NULL, key_data3.c_str(), key_data3.length()+1);
+    EXPECT_EQ(keys_container->store_key(key3), 0);
+    ASSERT_TRUE(keys_container->get_number_of_keys() == 3);
+
+    Key latest_percona_binlog_key("percona_binlog:2", NULL, NULL, NULL, 0);
+    ASSERT_TRUE(keys_container->remove_key(&latest_percona_binlog_key) == TRUE);
+    ASSERT_TRUE(keys_container->get_number_of_keys() == 3);
+
+    Key percona_binlog_key("percona_binlog", NULL, NULL, NULL, 0);
+    ASSERT_TRUE(keys_container->remove_key(&percona_binlog_key) == TRUE);
+    ASSERT_TRUE(keys_container->get_number_of_keys() == 3);
+
+    delete sample_key; // unused in this test
+  }
+
+  TEST_F(Keys_container_test, StoreStoreStoreRemoveFetchSystemKeyFetchRegularKey)
+  {
+    IKeyring_io *keyring_io= new Buffered_file_io(logger);
+    EXPECT_EQ(keys_container->init(keyring_io, file_name), 0);
+
+    std::string key_data1("system_key_data_1");
+    Key *key1= new Key("percona_binlog:0", "AES", NULL, key_data1.c_str(), key_data1.length()+1);
+
+    EXPECT_EQ(keys_container->store_key(key1), 0);
+    ASSERT_TRUE(keys_container->get_number_of_keys() == 1);
+
+    std::string key_data2("system_key_data_2");
+    Key *key2= new Key("percona_binlog:1", "AES", NULL, key_data2.c_str(), key_data2.length()+1);
+    key2->xor_data();
+    EXPECT_EQ(keys_container->store_key(key2), 0);
+    ASSERT_TRUE(keys_container->get_number_of_keys() == 2);
+
+    std::string key_data3("Robi3");
+    Key *key3= new Key("Roberts_key3", "AES", "Robert", key_data3.c_str(), key_data3.length()+1);
+    EXPECT_EQ(keys_container->store_key(key3), 0);
+    ASSERT_TRUE(keys_container->get_number_of_keys() == 3);
+
+    std::string key_data4("Robi4");
+    Key *key4= new Key("Roberts_key4", "AES", "Robert", key_data4.c_str(), key_data4.length()+1);
```

Comment:
Spaces around +.

I'll stop here, github is already slowing down

Reply:
All done